src/mission_start.cpp

```diff
@@ -614,6 +614,987 @@ void mission_start::radio_repeater( mission *miss )
+ bay.load(site.x * 2, site.y * 2, g->get_levz(), false);
+ bay.place_items( "mischw", 65, 2, 10, 4, 10, true, 0 );
+ bay.place_items( "mischw", 65, 2, 13, 4, 13, true, 0 );
+ bay.furn_set( 1, 15, f_fridge);
+ bay.spawn_item( 2, 15, "hdframe" );
+ bay.furn_set(3,15,f_washer);
+ bay.save();
+}
+
+void mission_start::ranch_bartender_1(mission *miss)
+{
+ npc *p = g->find_npc( miss->npc_id );
+ p->my_fac->wealth += rng(500,2500);
+ p->toggle_mutation( "NPC_BRANDY" );
+
+ tripoint site = target_om_ter_random("ranch_camp_51", 1, miss, false);
+ tinymap bay;
+ bay.load(site.x * 2, site.y * 2, g->get_levz(), false);
+ bay.draw_square_ter( t_wall_half, 1,15,7,17);
+ bay.draw_square_ter( t_dirt, 2,16,6,17);
+ bay.draw_square_ter( t_wall_half, 0,8,14,15);
+ bay.draw_square_ter( t_dirt, 1, 9, 13, 14);
+ bay.ter_set( 0, 10, t_door_frame);
+ bay.ter_set( 0, 11, t_door_frame);
+ bay.draw_square_ter( t_dirtfloor, 3, 15, 5, 15);
+ bay.draw_square_ter( t_dirtfloor, 3, 18, 5, 18);
+ bay.save();
+}
+
+void mission_start::ranch_bartender_2(mission *miss)
+{
+ npc *p = g->find_npc( miss->npc_id );
+ p->my_fac->wealth += rng(500,2500);
+ p->toggle_mutation( "NPC_RUM" );
+
+ tripoint site = target_om_ter_random("ranch_camp_51", 1, miss, false);
+ tinymap bay;
+ bay.load(site.x * 2, site.y * 2, g->get_levz(), false);
+ bay.draw_square_ter( t_wall_half, 14,10,19,15);
+ bay.draw_square_ter( t_dirt, 15,11,18,14);
+ bay.draw_square_ter( t_wall_half, 14,15,17,18);
+ bay.draw_square_ter( t_dirt, 15,15,16,18);;
+ bay.translate( t_door_frame, t_door_c);
+ bay.translate( t_wall_half, t_wall_wood);
+ bay.draw_square_ter( t_window_frame, 0, 13, 0, 13);
+ bay.draw_square_ter( t_window_frame, 4, 8, 5, 8);
+ bay.draw_square_ter( t_window_frame, 9, 8, 10, 8);
+ bay.draw_square_ter( t_window_frame, 19, 12, 19, 12);
+ bay.furn_set( 18, 19, f_wood_keg);
+ bay.furn_set( 16, 19, f_null);
+ bay.save();
+}
+
+void mission_start::ranch_bartender_3(mission *miss)
+{
+ npc *p = g->find_npc( miss->npc_id );
+ p->my_fac->wealth += rng(500,2500);
+ p->toggle_mutation( "NPC_WHISKEY" );
+
+ tripoint site = target_om_ter_random("ranch_camp_51", 1, miss, false);
+ tinymap bay;
+ bay.load(site.x * 2, site.y * 2, g->get_levz(), false);
+ bay.translate( t_window_frame, t_window_boarded_noglass);
+ bay.draw_square_ter( t_dirtfloor, 15,11,18,14);
+ bay.draw_square_ter( t_dirtfloor, 15,15,16,18);
+ bay.draw_square_ter( t_dirtfloor, 1, 9, 13, 14);
+ bay.draw_square_ter( t_dirtfloor, 15,11,18,14);
+ bay.save();
+}
+
+void mission_start::ranch_bartender_4(mission *miss)
+{
+ npc *p = g->find_npc( miss->npc_id );
+ p->my_fac->wealth += rng(500,2500);
+
+ tripoint site = target_om_ter_random("ranch_camp_51", 1, miss, false);
+ tinymap bay;
+ bay.load(site.x * 2, site.y * 2, g->get_levz(), false);
+ bay.draw_square_furn(f_standing_tank, 16, 11, 17, 11);
+ bay.draw_square_furn(f_fvat_empty, 17, 14, 18, 14);
+ bay.draw_square_furn(f_standing_tank, 13, 23, 13, 23);
```

Comment:
You don't really need to create variable `f_standing_tank` to use that function, that function can be used with the string id:
```C++
bay.draw_square_furn("f_standing_tank", 13, 23, 13, 23);
// Or (because it's just one square anyway):
bay.furn_set("f_standing_tank", 13, 23);
// Or (if there is a function that only takes the integer id), translate it just when needed:
bay.furn_set(furnfind("f_standing_tank"), 13, 23);
```

That avoids the changes to mapdata.h which requires (needlessly) re-compiling many files.